src/engine/target.js

```diff
@@ -314,8 +319,12 @@ class Target extends EventEmitter {
      */
     deleteVariable (id) {
         if (this.variables.hasOwnProperty(id)) {
+            const variable = this.variables[id];
```

Comment:
This is a minor thing, but I think holding on to a "variable" object that gets deleted feels like it could lead to some strange bugs/brain confusion editing this code in the future. If all you need is the name / cloudyness, maybe just take out those fields as "deletedVarName" and "deletedVariableWasCloud" or something?

Reply:
Makes sense!